### DISCUSSION TITLE
How to specify an if condition in the Elsa 3.0 Designer If activity and add an activity in "then" or "else"?

### DISCUSSION THREAD

**Discussion by shxylgithub** [elsa-workflows/elsa-core#5676] [General]
How to specify an if condition in the Elsa 3.0 Designer If activity and add an activity in "then" or "else"?

**Comment by glime-ai** [comment]
To specify an if condition in the Elsa 3.0 Designer If activity and add an activity in "then" or "else", you would typically follow these steps:

1. Add the `If` activity to your workflow in the Elsa Designer.
2. Set the `Condition` property of the `If` activity to the condition you want to evaluate.
3. For the `Then` property, add the activity (or activities) that you want to execute if the condition evaluates to `true`.
4. For the `Else` property, add the activity (or activities) that you want to execute if the condition evaluates to `false`.

However, based on the context provided, it seems there might be an issue with the Elsa 3.1.3 Designer where the `Then` and `Else` properties are not visible in the designer interface. This is indicated by the GitHub issue where a user asks why the `If` activity only has the `Condition` attribute and not the `Then` and `Else` attributes.

If you are facing this issue, it might be a bug or a limitation in the current version of the Elsa Designer. You may need to define the `Then` and `Else` branches programmatically in code, as shown in the first context source, or wait for an update that resolves this issue.

If you are defining the workflow in code, you can follow the example provided in the first context source:

```csharp
var workflow = new Workflow
{
    Root = new If
    {
        Condition = new(true), // Your condition here
        Then = new WriteLine("Hello World!"), // Activity to execute if condition is true
        Else = new WriteLine("Goodbye cruel world!") // Activity to execute if condition is false
    }
};
```

In this code snippet, `Condition` is set to `new(true)` as an example, and the `Then` and `Else` properties are assigned `WriteLine` activities with different messages.

Please note that if you are using the designer interface and the `Then` and `Else` properties are not available, you may need to check for updates or reach out to the Elsa support community for assistance.

This comment was generated by [Glime](https://glimelab.ai/?utm_source=dRbvTb7IhCSYiI&utm_medium=github).
